### PR TITLE
feat(balance): Railgun rebalance

### PR DESCRIPTION
**Pull request by yay855**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [ ] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

The makeshift railgun attachment and weapons had some minor issues- the attachment was a bit overpowered, while the heavy rail rifle and military rail rifle both had the RELOAD_ONE flag despite having an ammo capacity of 20. The military rail rifle also felt a little unrealistic as a semi-automatic weapon used in the military.

## Describe the solution

The makeshift railgun has been split into the makeshift rifle accelerator and the makeshift pistol accelerator. Both have had their handling further reduced as well. The heavy rail rifle and ferromagnetic rail rifle have had their RELOAD_ONE flag removed, description modified slightly to express this, and the ferromagnetic rail rifle has had its damage increased to 30 as an amount between the makeshift and heavy. Finally, all three ferromagnetic rail weapons have had their TRAIL flag removed, as a trail of smoke renders your character unable to see when facing more than one weakish enemy.

### CODE REVIEW

**Comment by 9SineD** [comment]
May I suggest something similar to the RM11B Scout Rifle's two-shot burst? The four-shot auto seems a bit overkill for "designed for urban warfare situations where ammo is scarce" and the 20-round magazine.

**Comment by yay855** [comment]
Oh, good idea. I'll do that.

File: data/json/items/gunmod/barrel.json [hunk 1]
```diff
@@ -59,14 +59,25 @@
...
     "dispersion_modifier": 25,
     "loudness_modifier": 15,
-    "damage_modifier": { "damage_type": "bullet", "amount": 15 },
-    "ups_charges_modifier": 30,
+    "damage_modifier": { "damage_type": "bullet", "amount": 10 },
+    "ups_charges_modifier": 20,
     "min_skills": [ [ "electronics", 4 ] ]
+  },
+  {
+    "id": "accelerator_pistol_gunmod",
```
Comment (chaosvolt) [inline data/json/items/gunmod/barrel.json:73]: Will want a recipe for this version too.

Comment (yay855) [reply data/json/items/gunmod/barrel.json:73]: Right, doy. Little out of sorts today, for understandable reasons. I'll get on that.

**Review by chaosvolt — APPROVED** [review]

**Pull request merged**